src/Action/Server.hs

```diff
@@ -205,10 +205,10 @@ showResults local links haddock args query results = do
             "&filter=" ++ intercalate "|" (mapMaybe (fmap fst . targetModule) ts) ++
             "&precise=on"
 
-        add x = ("?" ++) $ intercalate "&amp;" $ map (joinPair "=") $
+        add x = ("?" ++) $ intercalate "&" $ map (joinPair "=") $
             case break ((==) "hoogle" . fst) args of
                 (a,[]) -> a ++ [("hoogle", x)]
-                (a,(_,x1):b) -> a ++ [("hoogle", x1 ++ " " ++ x)] ++ b
+                (a,(_,x1):b) -> a ++ [("hoogle", escapeURL x1 ++ " " ++ x)] ++ b
```

Comment:
Why isn't this `escapeURL $ x1 ++ " " ++ x2`?

Reply:
Good question.

I think it's safe since `x1` is user input whereas as far as I can tell `x` is composed of strings w/o special characters -- assuming the database contains no packages with special characters in their names (and those should be invalid afaik).

But it wouldn't hurt to escape it anyway; in the line above then as well.

Would you like me to change it or leave it as it is?